conftest.py

```diff
@@ -215,6 +227,14 @@ def cleanup_hosts():
 
     cleanup_hosts()
 
+@pytest.fixture(scope='session')
+def pools_hosts_by_name_or_ip(hosts: list[Host]) -> Generator[dict[HostAddress, Host]]:
+    """All hosts of all pools, each indexed by their hostname_or_ip."""
+    yield {host.hostname_or_ip: host
+           for pool_master in hosts
+           for host in pool_master.pool.hosts
+           }
```

Comment:
Waiting for a code formatter to have this :angel::
```python
    yield {
        host.hostname_or_ip: host
        for pool_master in hosts
        for host in pool_master.pool.hosts
    }
```

Reply:
:thinking: what would it bring, aside from wasted vertical space?

Reply:
Well IMHO:
- This fixes the alignment asymmetry of `{` and `}`
- Fix the wasted horizontal space :wink: 
- Fix the indentation of the code, not a multiple of 4

Reply:
> I think explicit for loops would be more readable here

IIRC they would be less efficient